### DISCUSSION TITLE
Migrate to Gateway API

### DISCUSSION THREAD

**Discussion by musklor** [kubernetes-sigs/gateway-api#3027] [Q&A]
Hi,
Thanks a lot for your work.

I'd like to migrate from httpproxy to httproute.

Do you know how i can do that?

I'd like to have a A+ score on ssllabs 😀

I'd need hsts (Content-Security-Policy), rewrite caldav for nextcloud and Permissions-Policy.

apiVersion: projectcontour.io/vi
kind: HTTPProxy
metadata:
name: example
spec:
virtualhost:
fqdn: example.com
routes:

services:
name: app
port: 80
responseHeadersPolicy:
set:
name: Content-Security-Policy
value: "default-src 'self'"
name: Strict-Transport-Security
value: max-age=31536000; includeSubDomains

**Comment by youngnick** [comment]
@musklor, thanks for you question, but this repo is for development on Gateway API itself.

For your question, I think you'd be better to ask either the Contour team in #contour on Kubernetes Slack, or at github.com/projectcontour/contour. Or to look into github.com/kubernetes-sigs/ingress2gateway, which is designed to do what you're talking about (although I'm not sure where the HTTPProxy support is at right now).
